tools/customLinters/rubocop_consistent_quotation_mark.rb

```diff
@@ -0,0 +1,21 @@
+module CustomCops
+  class ConsistentQuotationMark < RuboCop::Cop::Base
+    NOT_ALLOWED_CHARACTERS = [
+      '“', # U+201C Left Double Quotation Mark
+      '”', # U+201D Right Double Quotation Mark
+      '‘', # U+2018 Left Single Quotation Mark
+      '’', # U+2019 Right Single Quotation Mark
+    ].freeze
+
+    NOT_ALLOWED_REGEX = /[#{NOT_ALLOWED_CHARACTERS.join}]/
```

Comment:
I tried briefly to have a constants file that both custom linters take from. I wasn't able to configure it properly –– guidance here would be great! Would I need to add something to the `application.rb` file?

Reply:
I believe you are in plain-old-ruby land, so you shouldn't need to touch application.rb. but you could extract the shared constants into any shared ruby file which you then required into both places.

Reply:
you will need to add this line before you can include the constants file:
```ruby
require_relative 'linter_constants'
```

application.rb is part of rails. my assumption is that the linter runs outside of rails, so it doesn't care what's in application.rb. instead, you have to use pure ruby to access the file.

in rails, everything within the rails directories is `require`d for you. outside of rails (or in rails and when accessing things that are outside of rails) you must `require` a file before you can access it. `include` is not a substitute for `require`, in fact once you are requiring, you could drop the `include` and just use `LinterConstants::NOT_ALLOWED_CHARACTERS` to access your constant.

clear as mud?

Reply:
Wow, that was very helpful, thank you! Got it working, tested on both "bad" ruby and haml files.